crates/polars-arrow/src/bitmap/iterator.rs

```diff
@@ -1,7 +1,30 @@
+use polars_utils::slice::load_padded_le_u64;
+
 use super::bitmask::BitMask;
 use super::Bitmap;
 use crate::trusted_len::TrustedLen;
 
+/// Calculates how many iterations are remaining, assuming:
+///  - We have length elements left.
+///  - We need max(consume, min_length_for_iter) elements to start a new iteration.
+///  - On each iteration we consume the given amount of elements.
+fn calc_iters_remaining(length: usize, min_length_for_iter: usize, consume: usize) -> usize {
+    let min_length_for_iter = min_length_for_iter.max(consume);
+    if length < min_length_for_iter {
+        return 0;
+    }
+
+    let obvious_part = length - min_length_for_iter;
+    let obvious_iters = obvious_part / consume;
+    // let obvious_part_remaining = obvious_part % consume;
+    // let total_remaining = min_length_for_iter + obvious_part_remaining;
+    // assert!(total_remaining >= min_length_for_iter);          // We have at least 1 more iter.
+    // assert!(obvious_part_remaining < consume);                // Basic modulo property.
```

Comment:
Are these accidental comments or superfluous?

Reply:
Neither, it's a mathematical proof written in the style of Rust code that it's correct.

Reply:
Ah.. check.